annotations.md

```diff
@@ -0,0 +1,120 @@
+# Attributes syntax
+
+We're currently using `@` and some ad hoc following stuff.
+
+Proposed changes:
+
+- Inner vs outer attribute semantics
+  - Attachment in AST
+- Inner attributes with no identifier
+- Tool attributes
+
+## Proposal
+
+Goal: to be flexible and extensible.
+
+### Syntax
+
+Inner/outer is not meant to be user-facing jargon.
+
+```
+item ::= ... | fn_decl | const_decl | import | inner_attr | outer_attr item
+stmt ::= ... | const_decl | outer_attr stmt
+
+inner_attr ::= `@` `(` (annot_item,)* `)`
```

Comment:
The lack of identifier after the `@` for inner attributes makes it awkward to use many, unrelated attributes for an item. Instead of them looking like function calls, they would need to be differentiated by keyword arguments. An example I'm thinking about would be in Rust `#[derive(Clone, Serialize)]` and `#[serde(rename_all = "camelCase")]` on a struct. How would you do the equivalent here?

```kcl
@(derive = [Clone, Serialize], serde = { rename_all = camelCase })
```

I know KCL is very different from Rust, so the specifics don't apply. But in most languages that have item attributes, it's common to want to apply multiple, unrelated attributes to the same declaration.

I think I'd prefer a different token or tokens the way Rust does it with `#` vs. `#!`.